Yank/yamlbuild.py

```diff
@@ -1177,14 +1178,17 @@ class YamlBuilder:
     def yank_options(self):
         return self._isolate_yank_options(self.options)
 
-    def __init__(self, yaml_source=None):
+    def __init__(self, yaml_source=None, platform_name=None):
```

Comment:
As far as concern `YamlBuilder`, I don't think there's a way around it. If we don't want the user to specify an option to it through the YAML script, we have to pass it as a keyword somewhere else in the API. I don't think it's going to be many anyway (still talking about `YamlBuilder` only) since most can be specified in the YAML script.

I agree with you and I'm a little uneasy sometimes with how we handle the myriad of options in the API. We can talk about it in the next huddle maybe?